lib/dest/write-contents/write-buffer.js

```diff
@@ -1,14 +1,30 @@
 'use strict';
 
+var iconv = require('iconv-lite');
+
 var fo = require('../../file-operations');
+var getCodec = require('../../codecs');
 
 function writeBuffer(file, optResolver, onWritten) {
+  var encoding = optResolver.resolve('encoding', file);
+  var codec = getCodec(encoding);
```

Comment:
Can `getCodec` (or a variation on it) be used as the "type" in the options instead of resolving a string then using this logic?